x/wasm/keeper/msg_dispatcher.go

```diff
@@ -155,6 +159,22 @@ func (d MessageDispatcher) DispatchSubmessages(ctx sdk.Context, contractAddr sdk
 	return rsp, nil
 }
 
+func logError(ctx sdk.Context, msg string, err error) {
+	logger := ctx.Logger()
+	if logger != nil {
+		logger.Info(msg, "error", err.Error())
+	}
+}
+
+func redactError(err error) string {
+	// FIXME: do we want to hardcode some constant string mappings here as well?
+	// Or better document them? (SDK error string may change on a patch release to fix wording)
+	// sdk/11 is out of gas
+	// sdk/5 is insufficient funds (on bank send)
+	codespace, code, _ := sdkerrors.ABCIInfo(err, false)
+	return fmt.Sprintf("Error: %s/%d", codespace, code)
```

Comment:
Do we need this "Error: " prefix because this string is embedded in the `Err` case of the result, so this would come from the containing structure?

I wonder if `codespace: %s, code: %d` would be more helpful to understand where the values comes from. Having the variable names available is how devs interact with those values at the Tendermint RPC and CosmJS level.

Reply:
I agree with @webmaster128 that `codespace: %s, code: %d` is more readable for humans but `codespace/code` is easier to parse. In both cases, I would expect people to ask what the data means. 

IMHO, let's start with `codespace/code` to optimize for parsing. We can help the devs with some doc page for common codes.
Hardcoding some constants would add complexity on the contract side as they may have to match for certain constants as well as the "codespace/code" types.

"Error: " as a prefix is a redundant information. A prefix may make sense though if we want to have different structures/ versions to be parseable, like
`e1/sdk/11`,`e2/sdk/11/some-more-info`, `e3/sdk/11/some-other-info/123`, `e4/sdk/11/{"some":"json"}`

Getting a bit crazy on this, we could also make the whole error message json so that it is easy to parse, better readable for humans and simpler to extend. `{"codespace":"sdk", "code: "11"}`

Reply:
Shouldn't parsability be a non-goal if we say there is no stability guarantee for the content of those error strings? The more we make them look like easy to parse, the mode users make assumptions on the format.